presto-main/src/main/java/io/prestosql/operator/scalar/timestamp/HumanReadableSeconds.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar.timestamp;
+
+import io.airlift.slice.Slice;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.type.StandardTypes;
+
+import java.util.StringJoiner;
+import java.util.concurrent.TimeUnit;
+
+import static io.airlift.slice.Slices.utf8Slice;
+
+@ScalarFunction("human_readable_seconds")
+public final class HumanReadableSeconds
+{
+    private static final String HUMAN_REDABLE_FORMAT = "%s, %s, %s, %s, %s";
+    private static final int DAYS_IN_WEEK = 7;
+    private static final String WEEK = "week";
+    private static final String DAY = "day";
+    private static final String HOUR = "hour";
+    private static final String MINUTE = "minute";
+    private static final String SECOND = "second";
+
+    private HumanReadableSeconds() {}
+
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice humanReadable(@SqlType(StandardTypes.DOUBLE) double inputSeconds)
+    {
+        long seconds = Double.valueOf(inputSeconds).longValue();
+
+        long weeks = processWeeks(seconds);
```

Comment:
I don't think the processXXX methods are necessary. This could be done more concisely, and IMO, easier to read and follow like this:

```java
long seconds = (long) inputSeconds;

long weeks = TimeUnit.SECONDS.toDays(seconds) / DAYS_IN_WEEK;
seconds -= TimeUnit.DAYS.toSeconds(weeks * DAYS_IN_WEEK);

long days = TimeUnit.SECONDS.toDays(seconds);
seconds -= TimeUnit.DAYS.toSeconds(days);

long hours = TimeUnit.SECONDS.toHours(seconds);
seconds -= TimeUnit.HOURS.toSeconds(hours);

long minutes = TimeUnit.SECONDS.toMinutes(seconds);
seconds -= TimeUnit.MINUTES.toSeconds(minutes);

return getTimePeriods(weeks, days, hours, minutes, seconds);
```

Reply:
It could be further simplified to this if you introduce constants from "number of seconds in a week", "number of seconds in a day", etc:

```java
long seconds = (long) inputSeconds;

long weeks = TimeUnit.SECONDS.toDays(seconds) / DAYS_IN_WEEK;
seconds = seconds % SECONDS_PER_WEEK;

long days = TimeUnit.SECONDS.toDays(seconds);
seconds = seconds % SECONDS_PER_DAY;

long hours = TimeUnit.SECONDS.toHours(seconds);
seconds = seconds % SECONDS_PER_HOUR;

long minutes = TimeUnit.SECONDS.toMinutes(seconds);
seconds = seconds % SECONDS_PER_MINUTE;

return getTimePeriods(weeks, days, hours, minutes, seconds);
```